code/src/java/pcgen/gui2/csheet/PreviewVariablesHandler.java

```diff
@@ -0,0 +1,80 @@
+package pcgen.gui2.csheet;
+
+import com.sun.webkit.dom.HTMLInputElementImpl;
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.w3c.dom.html.HTMLInputElement;
+import pcgen.facade.core.CharacterFacade;
+
+/**
+ * A {@link ChangeListener} that decorates a new document in context to automatically save inputs tagged with "target_var"
+ */
+public class PreviewVariablesHandler implements ChangeListener<Document>
+{
+    private CharacterFacade character;
+
+    public void setCharacter(CharacterFacade character)
+    {
+        this.character = character;
+    }
+
+    @Override
+    public void changed(ObservableValue<? extends Document> observableValue, Document oldDoc, Document newDoc)
+    {
+        if (newDoc == null)
+        {
+            return;
+        }
+        NodeList elements = newDoc.getElementsByTagName("input");
+        for (int i = 0; i < elements.getLength(); i++)
+        {
+            HTMLInputElementImpl element = (HTMLInputElementImpl) elements.item(i);
+            String key = getInputKey(element);
+            if (key == null || key.isEmpty())
+            {
+                continue;
+            }
+            setInputValue(element, character.getPreviewSheetVar(key));
+            element.addEventListener("change", evt ->
+            {
+                HTMLInputElement input1 = (HTMLInputElement) evt.getCurrentTarget();
```

Comment:
could just be called input?

Reply:
sure can.  i renamed the conflicting variable already